src/components/program/ProgramContentMaterialBlock.tsx

```diff
@@ -80,86 +78,91 @@ const ProgramContentMaterialBlock: React.VFC<{
               name: material.data?.name || formatMessage(programMessages.ProgramContentMaterialBlock.attachment),
             },
           }))
-          .map(material => (
-            <StyledMaterial
-              key={material.id}
-              className="mb-3"
-              onClick={async () => {
-                await insertMaterialAuditLog({
-                  variables: {
-                    data: { member_id: currentMemberId, target: material.id, action: 'download' },
-                  },
-                })
-                  .then(async res => {
-                    if (res.data?.insert_material_audit_log_one?.id) {
-                      if (material.data.url) {
-                        window.open(material.data.url)
-                        return
-                      }
-                      setIsDownloading(prev => ({ ...prev, [material.id]: true }))
-                      const materialLink = await getFileDownloadableLink(
-                        `materials/${appId}/${programContentId}_${material.data.name}`,
-                        authToken,
+          .map(
+            material =>
+              material.programContentId === programContentId && (
+                <StyledMaterial
+                  key={material.id}
+                  className="mb-3"
+                  onClick={async () => {
+                    await insertMaterialAuditLog({
+                      variables: {
+                        data: { member_id: currentMemberId, target: material.id, action: 'download' },
+                      },
+                    })
+                      .then(async res => {
+                        if (res.data?.insert_material_audit_log_one?.id) {
+                          if (material.data.url) {
+                            window.open(material.data.url)
```

Comment:
maybe should check url is legit 
url = new URL(string);

Reply:
I agree with you, adjusted.